test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/CpsPackageReferenceProjectTests.cs

```diff
@@ -4053,9 +4052,51 @@ public async Task GetInstalledAndTransitivePackagesAsync_ProjectReferenceWithSin
 
             // Assert
             Assert.Empty(result.InstalledPackages); // No installed packages
-            // Foreign transitive packages in assets file don't have dependencies.
-            // Those are not considered transitive packages
-            Assert.Empty(result.TransitivePackages);
+            Assert.Equal(1, result.TransitivePackages.Count);
+
+            var origins = result.TransitivePackages[0].TransitiveOrigins.Select(transitivePackage => transitivePackage.PackageIdentity.Id);
+            Assert.Contains("Project1", origins);
+        }
+
+        [Fact]
+        public async Task GetInstalledAndTransitivePackagesAsync_MutipleProjectReferencesWithSamePackage_EmptyInstalledAndTransitivePackagesAsync()
```

Comment:
Mentioned it earlier, https://github.com/NuGet/NuGet.Client/pull/5926#discussion_r1686934615, can we add a test where a project has a dependency coming through both project and a package.

Reply:
I added a test for this comment, let me know if I'm not addressing your concern

Reply:
Looks good.